akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -364,6 +420,19 @@ private[remote] class Decoder(
         }
       }
 
+      private def applyIncomingInstruments(decoded: InboundEnvelope, headerBuilder: HeaderBuilder): Unit = {
```

Comment:
Do we want to do this in the `Decoder` or the `Deserializer`. The difference is that when using parallel inbound lanes the `Deserializer` will be done in parallel, i.e. we prefer to do more things in the `Deserializer` than in the `Decoder`.

Another reason to move it to the `Deserializer` is that in the `Decoder` we have not yet deserialized the message, and the deserialized message may be of interest for the RemoteInstrument